ast.ts

```diff
@@ -40,6 +51,9 @@ export type Expr<A> =
   | {  a?: A, tag: "index", obj: Expr<A>, index: Expr<A> }
   | {  a?: A, tag: "method-call", obj: Expr<A>, method: string, arguments: Array<Expr<A>> }
   | {  a?: A, tag: "construct", name: string }
+  // array-expr should be plain format like 1, 2, 3 without brackets
+  // TODO: should we make use of AST nodes from list and tuple groups?
```

Comment:
eventually, I think you'll be dealing with `(1,2,3)` , `[1,2,3]`, and `1,2,3`. Am I right?

Reply:
Yes!